src/worker.rs

```diff
@@ -250,4 +243,17 @@ impl<T: Sync + Send + 'static> Worker<T> {
             }
         }
     }
+
+    fn prepare_new_matches(&mut self) {
```

Comment:
outlining seems reasonable but the name could be better (hardest problem :P).

How about: `reset_matches`? This doesn't really prepare any new matches. Instead it resets the matches when deleting (or using a negating pattern)

Reply:
done